src/status_im/common/enter_seed_phrase/view.cljs

```diff
@@ -51,43 +51,36 @@
       (string/replace #"\s+" " ")
       (string/trim)))
 
-(defn- recovery-form
-  [{:keys [seed-phrase word-count error-state? all-words-valid? on-change-seed-phrase
-           keyboard-shown? on-submit]}]
-  (let [button-disabled? (or error-state?
-                             (not (constants/seed-phrase-valid-length word-count))
-                             (not all-words-valid?))]
-    [rn/view {:style style/form-container}
-     [header word-count]
-     [rn/view {:style style/input-container}
-      [quo/recovery-phrase-input
-       {:accessibility-label      :passphrase-input
-        :placeholder              (i18n/label :t/seed-phrase-placeholder)
-        :placeholder-text-color   colors/white-opa-30
-        :auto-capitalize          :none
-        :auto-correct             false
-        :auto-focus               true
-        :mark-errors?             true
-        :word-limit               max-seed-phrase-length
-        :error-pred-current-word  partial-word-not-in-dictionary?
-        :error-pred-written-words word-not-in-dictionary?
-        :on-change-text           on-change-seed-phrase}
-       seed-phrase]]
-     [quo/button
-      {:container-style (style/continue-button keyboard-shown?)
-       :type            :primary
-       :disabled?       button-disabled?
-       :on-press        on-submit}
-      (i18n/label :t/continue)]]))
+(defn- recovery-phrase-form
+  [{:keys [title seed-phrase word-count on-change-seed-phrase on-input-ref]} & children]
+  (->> children
+       (into
+        [rn/view {:style style/form-container}
+         [header title word-count]
+         [rn/view {:style style/input-container}
+          [quo/recovery-phrase-input
+           {:accessibility-label      :passphrase-input
+            :on-input-ref             on-input-ref
+            :placeholder              (i18n/label :t/seed-phrase-placeholder)
+            :placeholder-text-color   colors/white-opa-30
+            :auto-capitalize          :none
+            :auto-correct             false
+            :auto-focus               true
+            :mark-errors?             true
+            :word-limit               max-seed-phrase-length
+            :error-pred-current-word  partial-word-not-in-dictionary?
+            :error-pred-written-words word-not-in-dictionary?
+            :on-change-text           on-change-seed-phrase}
+           seed-phrase]]])))
 
 (defn keyboard-suggestions
   [current-word]
   (->> mnemonic/dictionary
        (filter #(string/starts-with? % current-word))
        (take 7)))
 
-(defn screen
-  [recovering-keypair?]
+(defn recovery-phrase-screen
+  [{:keys [title recovering-keypair? render-controls]}]
```

Comment:
Here we're allowing for two new props to be passed to this `recovery-phrase-screen` component:
* the `title` prop for customising the title of the screen/form
* the `render-controls` prop for customising the submit button logic

Reply:
Maybe this component should be called the `recovery-phrase-view`, thoughts?

Reply:
It was created before we agreed that the components exposed should be called `view`. So maybe `view` is enough?

Reply:
Yeah `view` would be a good choice, unfortunately that name is already used in this namespace 😄 

Side note:
It seems this namespace exports a `view` component, a `screen` component (added by this PR), and this `recovery-phrase-screen` (maybe this should be private?). And what's interesting is that the `view` component is used by the screens namespace, and the `screen` component is used by the `view` component.

Ideally I would like components named `screen` to be used by the screens namespace, and components named `view` to be used by screens and other components. Thoughts?

Reply:
Sorry for the late response here @seanstrom 

Well, the convention is to always name `view` to the main component. But we should provide a better name for `screen` and the `recovert-phrase-screen` components.

It'd be great if you pick new names that are more descriptive or at least that remove the word `screen`.

I know it may be out of scope, so it's ok if you don't want to do the refactor here :+1: